tests/test_utils.py

```diff
@@ -0,0 +1,360 @@
+    raise paramiko.SSHException("Could not connect")
+
+class TestFedimgUtils(unittest.TestCase):
+
+    class MockPopen(object):
+        def __init(self):
+            pass
+
+        def communicate(self, input=None):
+            pass
+
+        @property
+        def returncode(self):
+            pass
+
+    def test_get_file_arch_x86_64(self):
+        filename = 'Fedora-Atomic-26-1.5.x86_64.raw.xz'
+        arch = fedimg.utils.get_file_arch(filename)
+        assert arch == 'x86_64'
+
+    def test_get_file_arch_aarch64(self):
+        filename = 'Fedora-Cloud-Base-26-1.5.aarch64.raw.xz'
+        arch = fedimg.utils.get_file_arch(filename)
+        assert arch == None
+
+    def test_get_rawxz_urls(self):
+        images = [
+            {
+                "arch": "x86_64",
+                "format": "qcow2",
+                "implant_md5": None,
+                "mtime": 1499295621,
+                "path": "CloudImages/x86_64/images/Fedora-Atomic-26-1.5.x86_64.qcow2",
+                "size": 702779904,
+                "subvariant": "Atomic",
+                "type": "qcow2",
+                "volume_id": None
+            },
+            {
+                "arch": "x86_64",
+                "format": "raw.xz",
+                "implant_md5": None,
+                "mtime": 1499295718,
+                "path": "CloudImages/x86_64/images/Fedora-Atomic-26-1.5.x86_64.raw.xz",
+                "size": 529608216,
+                "subvariant": "Atomic",
+                "type": "raw-xz",
+                "volume_id": None
+            },
+            {
+                "arch": "x86_64",
+                "bootable": False,
+                "disc_count": 1,
+                "disc_number": 1,
+                "format": "raw.xz",
+                "implant_md5": None,
+                "mtime": 1499291771,
+                "path": "CloudImages/x86_64/images/Fedora-Cloud-Base-26-1.5.x86_64.raw.xz",
+                "size": 154897200,
+                "subvariant": "Cloud_Base",
+                "type": "raw-xz",
+                "volume_id": None
+            },
+            {
+                "arch": "x86_64",
+                "format": "vagrant-libvirt.box",
+                "implant_md5": None,
+                "mtime": 1499291717,
+                "path": "CloudImages/x86_64/images/Fedora-Cloud-Base-Vagrant-26-1.5.x86_64.vagrant-libvirt.box",
+                "size": 229231805,
+                "subvariant": "Cloud_Base",
+                "type": "vagrant-libvirt",
+                "volume_id": None
+            },
+        ]
+
+        urls = fedimg.utils.get_rawxz_urls('https://fedoraproject.org', images)
+        atomic_url = 'https://fedoraproject.org/CloudImages/x86_64/images/Fedora-Atomic-26-1.5.x86_64.raw.xz'
+        cloud_base_url = 'https://fedoraproject.org/CloudImages/x86_64/images/Fedora-Cloud-Base-26-1.5.x86_64.raw.xz'
+
+        assert len(urls) == 2
+        assert atomic_url in urls
+        assert cloud_base_url in urls
```

Comment:
Is ```urls``` just a list of these two URLs? If so, you can use set comparison to make a bit stronger of an assertion:

```
self.assertEqual(set(urls), {atomic_url, cloud_base_url})
```